jira-config/jira-config-ui/src/main/resources/JIRA/JIRAAuth/BasicAuthConfigDisplayer.xml

```diff
@@ -0,0 +1,108 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.5" reference="JIRA.JIRAAuth.BasicAuthConfigDisplayer" locale="">
+  <web>JIRA.JIRAAuth</web>
+  <name>BasicAuthConfigDisplayer</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title/>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+#set ($basicAuthClass = 'JIRA.JIRAAuth.BasicAuthClass')
+#set ($confDoc = $xwiki.getDocument('JIRA.JIRAAuth.BasicAuthConfiguration'))
+#set ($confObjects = $confDoc.getObjects($basicAuthClass))
+#if ($xcontext.action == 'get')
+  #if ($request.getParameter('save') == 'true')
+    ## Save the changes
+    #if ($services.csrf.isTokenValid($!request.form_token))
+      #foreach ($obj in $confObjects)
```

Comment:
I have a doubt: $obj cannot be null with that API call?
Also I don't really understand the logic of that loop: AFAICS below you only display the form for a unique object, so you shouldn't have to save multiple objects?

Reply:
> I have a doubt: $obj cannot be null with that API call?

Good question, I checked and it seem that a new list is created by removing all null so it seem good.

> Also I don't really understand the logic of that loop: AFAICS below you only display the form for a unique object, so you shouldn't have to save multiple objects?

Well, to display the object we call the displayer for each config, so the displayer return the content only for one object. But for the save it's a bit different because we return the full form with all data the same time, so we call the displayer only once for all objects.

Reply:
> Not super fan of this design, IMO save should probably be done in another place

Which place would it be better ? I can move this code into an other page if you prefer ?